### PR TITLE
Refactor backup restoration validation handling and add new translations

### PR DESCRIPTION
**Pull request by stephdl**
This pull request includes a refactoring of the backup restoration process to handle validation errors more effectively. Additionally, new translations have been added to improve error messaging during backup restoration.

A function to clean the form when cancel or close has been added

REFs:
- https://github.com/NethServer/nethsecurity/pull/586
- https://github.com/NethServer/nethsecurity/issues/587

Capture d’écran du 2024-06-13 12-32-43

Capture d’écran du 2024-06-13 12-32-35

### CODE REVIEW

File: src/components/standalone/backup_and_restore/RestoreContent.vue [hunk 1]
```diff
@@ -186,9 +186,26 @@ async function restoreBackup() {
         errorRestoreBackup.value.notificationTitle = t('error.cannot_restore_backup')
       }
     } catch (exception: any) {
-      errorPage.value.notificationTitle = t('error.cannot_restore_backup')
-      errorPage.value.notificationDescription = t(getAxiosErrorMessage(exception))
-      errorPage.value.notificationDetails = exception.toString()
+      if (exception instanceof ValidationError) {
```
Comment (andre8244) [inline src/components/standalone/backup_and_restore/RestoreContent.vue:189]: - Validation errors (e.g. invalid passphrase) should be displayed with the red message under the invalid field, without showing the error notification. Runtime errors (e.g. restore failed) should be displayed only with the error notification.
- The validation error can be managed like this: https://github.com/NethServer/nethsecurity-ui/blob/main/src/components/standalone/firewall/nat/CreateOrEditNetmapRuleDrawer.vue#L268-L273
- Let's display under the passhprase field "Incorrect passphrase or invalid backup file" (use this string for the i18n keyfor "decryption_failed").

**Review by Tbaile — APPROVED** [review]

File: src/components/standalone/backup_and_restore/RestoreContent.vue [hunk 2]
```diff
@@ -186,9 +208,16 @@ async function restoreBackup() {
...
+        for (const [, value] of exception.errorBag) {
+          const errorMessage = value[0]
+          errorRestore.value.passphrase = t('error.backup_passphrase_' + errorMessage)
+        }
+      } else {
+        // this is an unknow global error not expected
```
Comment (andre8244) [inline src/components/standalone/backup_and_restore/RestoreContent.vue:217]:
```suggestion
        // this is an unknown error not expected
```

File: public/i18n/en/translation.json [hunk 3]
```diff
@@ -330,7 +330,9 @@
     "cannot_remove_unit": "Cannot remove unit",
     "cannot_scan_network": "Cannot scan network",
     "cannot_retrieve_vpn_networks": "Cannot retrieve VPN networks",
-    "length_12_max": "Maximum 12 characters allowed"
+    "length_12_max": "Maximum 12 characters allowed",
+    "backup_passphrase_decryption_failed": "Decryption failed, passphrase might be wrong",
```
Comment (andre8244) [inline public/i18n/en/translation.json:334]:
```suggestion
    "backup_passphrase_decryption_failed": "Decryption failed: passphrase may be incorrect",
```

File: public/i18n/en/translation.json [hunk 4]
```diff
@@ -330,7 +330,9 @@
...
+    "backup_passphrase_restore_failed": "Restore failed or passphrase might be missing"
```
Comment (andre8244) [inline public/i18n/en/translation.json:335]:
```suggestion
    "backup_passphrase_restore_failed": "Restore failed: passphrase may be required or backup file may be invalid"
```

**Review by andre8244 — COMMENTED** [review]
Just some rephrasing

**Review by gsanchietti — APPROVED** [review]

**Review by andre8244 — COMMENTED** [review]
LGTM, let's resolve the conflicts on translation strings

**Pull request merged**